src/NuGet.Core/NuGet.Build.Tasks/StaticGraphRestoreArguments.cs

```diff
@@ -63,7 +68,10 @@ public static StaticGraphRestoreArguments Read(Stream stream)
         /// <param name="stream">A <see cref="Stream" /> to write the arguments to.</param>
         public void Write(Stream stream)
         {
-            using BinaryWriter writer = new BinaryWriter(stream, Encoding.Default, leaveOpen: true);
+            using BinaryWriter writer = new BinaryWriter(stream, Encoding.UTF8, leaveOpen: true);
+
+            // Write a byte indicating the beginning of the content
+            writer.Write(byte.MaxValue);
```

Comment:
Is just `byte.MaxValue` future proof enough?
Probably write several predefined sequential bytes something like `!NuGetStartSignal!` as bytes instead of just 1 byte.

Reply:
Yes sorry completely disregard my first iteration of this pull request, I've updated it to actually look at the encoding preamble which is variable length and added unit tests.